### PR TITLE
Docs/tutorial solutions/exercises

### PR DESCRIPTION
**Pull request by jlnav**
experimenting with adding drop-down boxes for tutorial exercises/solutions

### CODE REVIEW

**Comment by jlnav** [comment]
Decided to leave exercises for only the first two tutorials for now, since the last two tutorials don't especially involve writing functions, only demonstrating capabilities

File: docs/advanced_installation.rst [hunk 1]
```diff
@@ -14,7 +14,7 @@ pip
 ---
 
 We always recommend installing in a virtual environment such as Conda.
-If not, then use ``pip3``, ``python3`` below.
```
Comment (shuds13) [inline docs/advanced_installation.rst:17]: I'm okay with python/pip without the 3. But we should be consistent. Later python3 is used in a couple of places (when running the calling script). We should decide generally for the docs whether we want to drop the 3.

File: docs/tutorials/local_sine_tutorial.rst [hunk 2]
```diff
@@ -286,15 +337,63 @@ script and run ``python3 calling_script.py`` again
...
+                     'out': [('y', float)]}
+
+       persis_info = add_unique_random_streams({}, nworkers+1)
+       exit_criteria = {'gen_max': 160}
+
+       H, persis_info, flag = libE(sim_specs, gen_specs, exit_criteria, persis_info,
+                                   libE_specs=libE_specs)
+
+       if flag != 0:
+          print('Oh no! An error occurred!')
```
Comment (shuds13) [inline docs/tutorials/local_sine_tutorial.rst:391]: What do people think about introducing parse_args here somewhere?

**Comment by shuds13** [comment]
Checking here - I think somewhere the tutorials should have a link/location where to find example in the repo. Maybe at bottom of first section. "This example can be found in the repository at...."

File: docs/tutorials/local_sine_tutorial.rst [hunk 3]
```diff
@@ -286,15 +337,63 @@ script and run ``python3 calling_script.py`` again
   plt.legend(loc = 'lower right')
   plt.savefig('tutorial_sines.png')
 
----
+Exercise
+^^^^^^^^
+
+Write a Calling Script with the following specifications:
+
+  1. Use the Exercise simulator and generator functions instead
```
[the remark this answers is not in the document]
Comment (jlnav) [reply docs/tutorials/local_sine_tutorial.rst:345]: As in, list everything, then say: "Write a Calling Script with the above specifications?"

[the remark this answers is not in the document]
Comment (shuds13) [reply docs/tutorials/local_sine_tutorial.rst:347]: I misinterpreted this line as saying instead do the following lines. Perhaps say something like use the functions from the answers to the exercises.

**Comment by shuds13** [comment]
Open to views but we may want to standardise the H argument coming in and the out as H_in and H_out maybe?

**Comment by shuds13** [comment]
Optional:
"Our simulator function is called by a worker for every value in its batch from the generator function." could maybe be
 "Our simulator function is called by a worker for every work item produced by the generator function"

**Review by shuds13 — APPROVED** [review]
I made a few comments. All optional.

**Pull request merged**